backend/main.cpp

```diff
@@ -0,0 +1,199 @@
+            }
+            std::cout << std::endl;
+        }
+
+        // AI did these, I have no idea if this is legit -Pat
+        // Send a response back to the frontend with CORS headers
+        // std::string response = "Matrices received successfully!";
+        std::string response = matrices.result_matrix.to_string();
+        boost::asio::streambuf response_buf;
+        std::ostream response_stream(&response_buf);
+        response_stream << "HTTP/1.1 200 OK\r\n";
+        response_stream << "Content-Type: text/plain\r\n";
+        response_stream << "Access-Control-Allow-Origin: *\r\n"; // Enable CORS for all origins
+        response_stream << "Content-Length: " << response.size() << "\r\n\r\n";
+        response_stream << response;
+        boost::asio::write(socket, response_buf);
+    } catch (const std::exception& e) {
+        std::cerr << "Error handling request: " << e.what() << std::endl;
+        // Send an error response back to the frontend
+        std::string response = "Error handling request: ";
+        response += e.what();
+        boost::asio::streambuf response_buf;
+        std::ostream response_stream(&response_buf);
+        response_stream << "HTTP/1.1 500 Internal Server Error\r\n";
+        response_stream << "Content-Type: text/plain\r\n";
+        response_stream << "Access-Control-Allow-Origin: *\r\n"; // Enable CORS for all origins
+        response_stream << "Content-Length: " << response.size() << "\r\n\r\n";
+        response_stream << response;
+        boost::asio::write(socket, response_buf);
+    }
+
+}
+
+// Completely AI-cooked, I have no idea what's going on here but it looks alright to me and it works
+void start_server() {
+    try {
+        // Create io_context
+        io_context io;
+
+        // Create an acceptor to listen for incoming connections
+        ip::tcp::acceptor acceptor(io, ip::tcp::endpoint(ip::tcp::v4(), POST_PORT));
+
+        std::cout << "Server started. Waiting for incoming connections..." << std::endl;
+
+        // Start accepting connections
+        while (true) {
+            // Create a socket for the incoming connection
+            ip::tcp::socket socket(io);
+
+            // Wait for a client to connect
+            acceptor.accept(socket);
+
+            // Read request data from the client
+            std::string request;
+            boost::system::error_code error;
+            size_t bytes_transferred;
+            do {
+                char buffer[1024]; // Adjust buffer size as needed
+                bytes_transferred = socket.read_some(boost::asio::buffer(buffer), error);
+                request.append(buffer, buffer + bytes_transferred);
+            } while (0);
+            // Idk why but the original thing chatgpt suggested was this with no loop and the requests would just be zero length?? 
+            // Then it suggested do-while(bytes_transferred > 0) but that would just hang forever so I just put 0 and it works fine
```

Comment:
this is the same as not having a loop tho, did it not work without a loop?

Reply:
I looked more closely, and I think the actual difference was that socket.read() didn’t work but socket.read_some() did... I'll play with it some more later but for now it works